sdk/core/Azure.Core/src/Internal/SpanOfCharsExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core
+{
+    internal static class SpanOfCharsExtensions
+    {
+        public static void AppendString(this ref Span<char> span, ref int offset, string str)
+        {
+#if NET6_0_OR_GREATER
+            str.CopyTo(span[offset..]);
+#else
+            str.AsSpan().CopyTo(span.Slice(offset));
+#endif
+            offset += str.Length;
+        }
+
+        public static void AppendChar(this ref Span<char> span, ref int offset, char value)
+        {
+            span[offset] = value;
```

Comment:
Is it the caller's responsibility to do bounds-checking, or would it make sense for us to add parameter validation?

Reply:
I would prefer caller to be responsible for bounds-checking. This is a very low-level extension method, caller must be aware of what it is doing.

Reply:
Whole type is removed in favor of `DefaultInterpolatedStringHandler`